dev/ci/gitlab-ci-ctests.yml

```
@@ -45,9 +53,54 @@
   extends: .base_config
   stage: run_tests
   script: |
-    set -e  # Fail the job if any command fails
-    echo "Running ${CTEST} tests in ${GW_HOMEgfs}/dev/ctests"
-    cd ${GW_HOMEgfs}/dev/ctests/build
-    ctest -L ${CTEST} --output-on-failure
+    echo "Using build directory ${GW_HOMEgfs}"
+    source "${GW_HOMEgfs}/dev/ush/gw_setup.sh"
+    cd "${GW_HOMEgfs}/dev/ctests/build" || exit 1
+    echo "Running CTests with label: ${CTEST_NAME} in ${PWD}"
+    num=$(ctest -N | grep "Total Tests" | awk "{print \$3}")
+    if [[ "${num}" -gt 0 ]]; then
+      echo "CTests found: ${num}"
+    else
+      echo "No tests found"
+      exit 1
+    fi
+    ctest -L "${CTEST_NAME}" --output-junit result.xml --stop-on-failure --output-on-failure
+    ctest_status=$?
+
+    # Check the JUnit XML file for failures explicitly
+    if [[ -f result.xml ]]; then
+      # Count failures in the JUnit XML
+      failures=$(grep -c "failures=\"[1-9][0-9]*\"" result.xml || true)
+      errors=$(grep -c "errors=\"[1-9][0-9]*\"" result.xml || true)
+
+      # If there are failures or errors in the JUnit report, fail the job
+      if [ "$failures" -gt 0 ] || [ "$errors" -gt 0 ] || [ "${ctest_status}" -ne 0 ]; then
```

I find it cleaner to put the `||` inside of the logical test.  Also, please use braces around variables:

```suggestion
      if [[ ${failures} -gt 0 || ${errors} -gt 0 || ${ctest_status} -ne 0 ]]; then